CodeJam.Main.Tests/Threading/TaskHelperTests.ForEachAsync.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using NUnit.Framework;
+
+#if NET45_OR_GREATER || TARGETS_NETCOREAPP
+using TaskEx = System.Threading.Tasks.Task;
+#elif NET40_OR_GREATER
+using TaskEx = System.Threading.Tasks.TaskEx;
+#else
+using TaskEx = System.Threading.Tasks.Task;
+#endif
```

Comment:
```suggestion
#if (NET45_OR_GREATER || TARGETS_NETCOREAPP) && !NET40_OR_GREATER
using TaskEx = System.Threading.Tasks.Task;
#else
using TaskEx = System.Threading.Tasks.TaskEx;
#endif
```

Reply:
I'd prefer to keep the conditional targeting as simple as possible. We have a lot of issues with improper targeting in the past and it is really a pain to troubleshoot it.

Reply:
Ok. But make it consistent .